### PR TITLE
[Enhancement] Increase the default identifier name length to 1023 

### PR DESCRIPTION
**Pull request by jiacheng-celonis**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes # N/A

## Problem Summary(Required) ：

Previously, the common identifier length limit is 63 characters (for DB names and aliases). We raised the limit to 1023 (the default limit in Hive) in this PR.

Note that for table name and column name, their limit is already increased to 1024: https://github.com/StarRocks/starrocks/pull/14896.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by jiacheng-celonis** [comment]
This is no longer necessary after https://github.com/StarRocks/starrocks/pull/14896.

**Comment by jiacheng-celonis** [comment]
> This is no longer necessary after #14896.

It turns out that we still need this because we are exceeding the identifier length limit for DB names.

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dengliu — COMMENTED** [review]
there was a **createViewBadName** test in this file before. shall we add it?

https://github.com/StarRocks/starrocks/blob/db1716d7b4659bd300ae514618d8211283d3956b/fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewTest.java#L279-L279

```
    @Test
    public void createViewBadName() {
        String longLongName = "view___123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890q";
        String sql = "create view db1." + longLongName + " as select 1,2,3";
        try {
            UtFrameUtils.parseStmtWithNewParser(sql, connectContext);
            Assert.fail(); // should raise Exception
        } catch (Exception e) {
            Assert.assertEquals("Incorrect table name '" + longLongName + "'", e.getMessage());
        }
    }
```

**Comment by jiacheng-celonis** [comment]
> there was a **createViewBadName** test in this file before. shall we add it?

Done. Added a test.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dengliu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_admit_test

**Comment by imay** [comment]
run starrocks_admit_test

**Comment by imay** [comment]
run starrocks_admit_test

**Comment by chaoyli** [comment]
@mergify backport branch-2.5

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateTableTest.java [hunk 1]
```diff
@@ -112,4 +114,17 @@ public void test1AutoTabletWithPartition() throws Exception {
         }
         Assert.assertEquals(bucketNum, 12);
     }
+
+
+    @Test
+    public void createBadDbName() {
+        String longDbName = new String(new char[1026]).replace('\0', 'a');
```
Comment (jaogoy) [inline fe/fe-core/src/test/java/com/starrocks/analysis/CreateTableTest.java:121]: It also needs a functional test, up to 1023 size db name.

**Pull request merged**